src/hw1/ParallelExternalLongSorter.java

```diff
@@ -238,11 +241,11 @@ private long getWorkingMemoryLimit() {
         //      looks like there's a 1MB overhead just for creating a thread
         //      https://dzone.com/articles/how-much-memory-does-a-java-thread-take
         // naive implementation - save some space just in case
-        final long overhead = 128 * MB;
-        //  Maximum heap size: "Smaller of 1/4th of the physical memory or 1GB"
+        final long overhead = 256 * MB;
```

Comment:
did this help with performance or is it arbitrary?

Reply:
at first i thought i wasnt leaving enough room on the heap for overhead, running some tests now to make sure this can be set lower... gonna set this back down before merging

Reply:
i can't even get it to throw a heap error when i give it only 1 MB for overhead lol (it actually ended up with 1.5MB in that test because of how the # of chunks worked out). it does crash with only 1 MB for overhead though (as expected)

the purpose of that overhead was for stuff allocated on the heap taking up space, e.g. pointers and stuff -- everything else can get paged in an out as needed from memory mapping

Reply:
so i realized probably the best way to run it out of memory is use 8 threads and watch the logs to try to game it (push 'expected memory use' as close to 'working memory' as possible)

i got it to crash with up to 4 MB of overhead with these values
vm args: -ea -Xmx385M
program args: 100000000.bin sorted.bin 8
overhead=1 * MB

now im incrementing both Xmx and overhead by the same amount (crashing up to overhead=12 MB so far)

Reply:
| threads | overhead | ✅ =pass ❌ =fail |
| -- | -- | -- |
| 2 | 4 MB | ❌ |
| 2 | 8 MB | ❌ |
| 2 | 40 MB | ❌ |
| 2 | 50 MB | ❌ |
| 2 | 57 MB | ❌ |
| 2 | 61 MB | ✅  |
| 2 | 70 MB | ✅  |
| 4 | 16 MB | ❌ |
| 4 | 24 MB | ❌ |
| 4 | 32 MB | ❌ |
| 4 | 60 MB | ❌ |
| 4 | 64 MB | ❌ |
| 4 | 69 MB | ❌ |
| 4 | 70 MB | ✅  |
| 8 | 16 MB | ❌ |
| 8 | 24 MB | ✅  |
|64 | 11 MB | ✅ |
| 64 | 15 MB | ✅ |
| 64 | 16 MB | ✅ |
| 128 | 16 MB | ✅ |
| 128 | 19 MB | ✅ |
| 128 | 24 MB | ✅ |

so... that's pretty weird. looks like memory use peaks at 4 cores on my machine. in any case, it looks like everything should sort with 70MB of overhead. I'm going to double it for a margin of safety